src/main/java/connectify/logic/commands/RankPersonCommand.java

```diff
@@ -0,0 +1,52 @@
+package connectify.logic.commands;
+
+import static connectify.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.Comparator;
+
+import connectify.model.Model;
+import connectify.model.person.Person;
+
+/**
+ * Lists all persons in the address book to the user according to their priority.
+ */
+public class RankPersonCommand extends Command {
+
+    public static final String COMMAND_WORD = "rankPerson";
+
+    public static final String MESSAGE_SUCCESS = "Ranked all persons";
+
+    public static final String EMPTY_LIST_MESSAGE = "There are no persons in Connectify";
+
+    @Override
+    public CommandResult execute(Model model) {
+        requireNonNull(model);
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_PERSONS);
+        Comparator<Person> comparator = Comparator.comparing(Person::rank);
+        model.updateSortedPersonList(comparator);
```

Comment:
Thanks for the changes! Looks good to me! However, can I check if running the usual "list" command will display the original list order of the addressbook? Seems like the "rank" command will permanently update the order.

Reply:
Great question! The 'rank' command does permanently update the order in the person list, however, the delete person and edit person command will still work, by using the new index to refer to the person to be deleted.